src/include/cell.i

```diff
@@ -72,6 +73,8 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
 
     __cell_check_value_validity(session, tw);
 
+    r = session->reconcile;
```

Comment:
__wt_bulk_insert_var() calls __wt_cell_pack_del() which calls __cell_pack_value_validity(). session->reconcile is not used in the call chain, instead (WT_CURSOR_BULK *)->r.

Assuming that __cell_check_value_validity() will be called with session->reconcile set seems problematic to me. One it doesn't seem to be always true. Two, it sets a hidden expectation that this function and its caller will have a session->reconciliation structure set.

I don't have a good solution. One could be passing a variable to __cell_pack_value_validity(,..., &stats), and then setting that variable to true if we want to collect stats. Other could be passing the reconciliation structure through the stack to this function.

Reply:
I have looked at the code and the cleanest way so far looks like passing reconcile structure through call chain.

Reply:
Let's do that then.